frontend/src/shared/auth/hook.ts

```diff
@@ -0,0 +1,41 @@
+import { useDispatch, useSelector } from "react-redux";
+
+import type { AppState } from "declarations";
+
+import { ConfirmEmailApiArg, LoginApiArg, useConfirmEmailMutation, useLoginMutation } from "../api/generated";
+import { type AuthSliceState, setLoggedIn, setLoggedOut } from "./slice";
+
+export const useAuth = () => {
+  const dispatch = useDispatch();
+  const auth: AuthSliceState = useSelector((state: AppState) => state.auth);
+  const [confirmEmailMutation, { isLoading: isConfirmEmailLoading }] = useConfirmEmailMutation();
+  const [loginMutation, { isLoading: isLoginLoading }] = useLoginMutation();
+
+  const confirmEmail = async (arg: ConfirmEmailApiArg) => {
```

Comment:
for future, it would be nice to declare return type for confirmEmail (): Promise<.....> =>